ui/src/linking/linking-api.ts

```diff
@@ -57,7 +44,15 @@ const LINKING_URI = `${API_URI}/linking`;
 
 const geometryElementsLinkedStatusCache = asyncCache<GeometryPlanId, GeometryPlanLinkStatus>();
 const suggestedSwitchesCache = asyncCache<string, SuggestedSwitch[]>();
-const locationTrackEndpointsCache = asyncCache<string, LocationTrackEndpoint[]>();
+
+type LinkingDataType = 'reference-lines' | 'location-tracks' | 'switches' | 'km-posts';
+type LinkingType = 'geometry' | 'empty-geometry' | 'suggested';
+
+function linkingUri(dataType: LinkingDataType, linkingType: LinkingType, id?: string): string {
```

Comment:
Tämä on vähän hyvä-ja-huono refaktorointi, koska urleja on vaikeampi hakea kun ne on tehty näin... mutta sain varmemmin tehtyä kaikki yhtenevästi ilman typoja kun värväsin TS:n apuun